apps/evm-profile/src/components/Profile/UploadProfileImage.tsx

```diff
@@ -0,0 +1,61 @@
+import { toast } from '@/hooks/useToast';
+import { ProfileMetadata } from '@/types';
+import { IPFS_GATEWAY } from '@/utils/constants';
+import { uploadProfileImage } from '@/utils/profileHelpers';
+import { ChangeEvent, SyntheticEvent, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import Uploader from '../Ui/Uploader';
+
+interface UploadProfileImageProps {
+  value: 'image' | 'profile.banner';
+  innerText: string | undefined;
+  errorMessage?: string;
+}
+
+export default function UploadProfileImage({
+  value,
+  innerText,
+  errorMessage,
+}: UploadProfileImageProps) {
+  const { setValue, trigger } = useFormContext<ProfileMetadata>();
+  const [fileName, setFileName] = useState('');
+  const [isUploadingImage, setIsUploadingImage] = useState(false);
+
+  const uploadImage = async (
+    e: SyntheticEvent | ChangeEvent<HTMLInputElement>,
+  ) => {
+    setIsUploadingImage(true);
+    const result = await uploadProfileImage(e);
+    if (typeof result !== 'string' && result?.hash && result?.fileName) {
```

Comment:
This can be cleaner:

`if (result && typeof result !== 'string') { ...`

The only other possibility is the hash so you don't need to check for each field in the hash